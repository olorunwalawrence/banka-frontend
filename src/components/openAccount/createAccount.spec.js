import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount, toJson } from '../../../test/enzyme';
import Form from './createForm';
import OpenAccountLayout from './openAccountLayout';

let user = {
  id: 2,
  firstname: 'Prince',
  lastname: 'mike',
  email: 'mike@email.com',
  type: 'user',
  isadmin: false,
}
const context = {
  auth: {
    isLoading: false,
    user,
    error: ''
  },
  account: {
    success: '',
    error: '',
    accountDetails: {}
  }
}
const store = {
  getState: () => ({ ... context }),
  subscribe: () => ({}),
  dispatch: () => ({
    type: CREATE_ACCOUNT, 
    payload: {
      success,
      error: '',
      accountDetails: {
        ...data
      }
    }
  })
}

describe('Layout Test', () => {
  let wrapper;
  it('should display the layout for creating an account properly', () => {
    wrapper = mount(
      <Router>
        <Provider store={store}>
          <OpenAccountLayout />
        </Provider>
      </Router>
    );
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should display the layout for creating an account properly', () => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <OpenAccountLayout />
        </Router>
      </Provider>
    );
    
    expect(wrapper.find('UserHeader')).toBeTruthy();
    expect(wrapper.find('.drop-menu')).toBeTruthy();
    expect(wrapper.find('OpenAccountForm')).toBeTruthy();
  });
  it('should display the layout for creating an account properly', () => {
    context.auth.user = {};
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <OpenAccountLayout />
        </Router>
      </Provider>
    );
    
    expect(wrapper.find('Redirect').props().to).toEqual('/login');
  });
});

describe('Form Test', () => {
  let wrapper;
  it('should display the layout for creating an account properly', () => {
    wrapper = mount(
        <Provider store={store}>
          <Form />
        </Provider>
    );
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should display the layout for creating an account properly', () => {
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Form />
        </Router>
      </Provider>
    );
    const option = wrapper.find('select').props().children;
    
    expect(wrapper.find('form')).toBeTruthy();
    expect(wrapper.find('select')).toBeTruthy();
    expect(wrapper.find('select').props().children.length).toEqual(3);
    expect(option[0].props.children).toEqual('Choose an account type');
    expect(option[1].props.children).toEqual('Current');
    expect(option[2].props.children).toEqual('Savings');
    expect(wrapper.find('label').props().children).toEqual('Account type');
    expect(wrapper.find('button').text()).toEqual('Create');
  });
  it('should display the layout for creating an account properly', () => {
    context.account.success = 'You account has been opened';
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Form />
        </Router>
      </Provider>
    );
    
    expect(wrapper.find('.alert-success').text()).toEqual(context.account.success);
  });
});
