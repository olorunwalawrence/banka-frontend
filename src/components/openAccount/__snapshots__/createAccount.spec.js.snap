// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Test should display the layout for creating an account properly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(OpenAccountForm)>
    <OpenAccountForm
      account={Object {}}
      error=""
      isLoading={false}
      openAccount={[Function]}
      success=""
    >
      <div
        className="form sm-grid"
      >
        <div
          className="container x"
        >
          <div
            className="dialog m"
          />
          <div
            className="form-card"
          >
            <h2>
              Open Account
            </h2>
            <div />
            <form
              onSubmit={[Function]}
            >
              <label>
                Account type
              </label>
              <div
                className="form-group"
              >
                <select
                  className="select"
                  defaultValue="Choose an account type"
                  id="select"
                  onChange={[Function]}
                >
                  <option
                    disabled={true}
                  >
                    Choose an account type
                  </option>
                  <option
                    value="current"
                  >
                    Current
                  </option>
                  <option
                    value="savings"
                  >
                    Savings
                  </option>
                </select>
              </div>
              <div
                className="form-group"
              >
                <button
                  className="btn btn-green"
                  id="createAccount"
                  type="submit"
                >
                  Create
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </OpenAccountForm>
  </Connect(OpenAccountForm)>
</Provider>
`;

exports[`Layout Test should display the layout for creating an account properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(Layout)>
        <Layout
          dispatch={[Function]}
          user={
            Object {
              "email": "mike@email.com",
              "firstname": "Prince",
              "id": 2,
              "isadmin": false,
              "lastname": "mike",
              "type": "user",
            }
          }
        >
          <Connect(UserHeader)>
            <UserHeader
              logout={[Function]}
            >
              <div
                className="container"
                id="heed"
              >
                <h1
                  className="white"
                >
                  <span>
                    <button
                      className="drop-butn nn hiden"
                      onMouseDown={[Function]}
                      type="button"
                    >
                      <i
                        className="icofont-navigation-menu icofont-2x"
                      />
                    </button>
                    MyBanka
                  </span>
                </h1>
              </div>
              <ul
                className=" drop-menu hide"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <Link
                  to="/home"
                >
                  <a
                    href="/home"
                    onClick={[Function]}
                  >
                    <li>
                      Dashboard
                    </li>
                  </a>
                </Link>
                <Link
                  to="/account"
                >
                  <a
                    href="/account"
                    onClick={[Function]}
                  >
                    <li>
                      Create Account
                    </li>
                  </a>
                </Link>
                <Link
                  onClick={[Function]}
                  to="/"
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <li>
                      Logout
                    </li>
                  </a>
                </Link>
              </ul>
            </UserHeader>
          </Connect(UserHeader)>
          <main
            className="flex"
          >
            <div
              className="one-third aside"
            >
              <ul
                className="user"
              >
                <li>
                  <Link
                    to="/home"
                  >
                    <a
                      href="/home"
                      onClick={[Function]}
                    >
                      Dashboard
                    </a>
                  </Link>
                </li>
                <li>
                  <Link
                    to="/account"
                  >
                    <a
                      href="/account"
                      onClick={[Function]}
                    >
                      Create Account
                    </a>
                  </Link>
                </li>
                <li>
                  <Link
                    onClick={[Function]}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Logout
                    </a>
                  </Link>
                </li>
              </ul>
            </div>
            <div
              className="two-third main"
            >
              <Connect(OpenAccountForm)>
                <OpenAccountForm
                  account={Object {}}
                  error=""
                  isLoading={false}
                  openAccount={[Function]}
                  success=""
                >
                  <div
                    className="form sm-grid"
                  >
                    <div
                      className="container x"
                    >
                      <div
                        className="dialog m"
                      />
                      <div
                        className="form-card"
                      >
                        <h2>
                          Open Account
                        </h2>
                        <div />
                        <form
                          onSubmit={[Function]}
                        >
                          <label>
                            Account type
                          </label>
                          <div
                            className="form-group"
                          >
                            <select
                              className="select"
                              defaultValue="Choose an account type"
                              id="select"
                              onChange={[Function]}
                            >
                              <option
                                disabled={true}
                              >
                                Choose an account type
                              </option>
                              <option
                                value="current"
                              >
                                Current
                              </option>
                              <option
                                value="savings"
                              >
                                Savings
                              </option>
                            </select>
                          </div>
                          <div
                            className="form-group"
                          >
                            <button
                              className="btn btn-green"
                              id="createAccount"
                              type="submit"
                            >
                              Create
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </OpenAccountForm>
              </Connect(OpenAccountForm)>
            </div>
          </main>
        </Layout>
      </Connect(Layout)>
    </Provider>
  </Router>
</BrowserRouter>
`;
